### PR TITLE
Update bitmaskRbac for more robust dynamic roles

### PR DESCRIPTION
**Pull request by nmclrn**
Deleting roles is not currently supported, and we still do not enforce a minimum count of admin users.

Also add the following fixes and checks:

 - Guard against duplicate roles
 - Change the "admin" role to "rbac_admin"
 - Make display names optional
 - Restrict role params to know roles to guard against typos and look-a-like string attacks
 - Track the constructing admin account properly so that it cannot be duplicated
 - Validate bitmask param
 - Add tests all of the above, including the "many roles" use case
 - constant -> view
 - Ignore build files
 - Report gas during tests

TODO: publish as an npm module, have TruSetRBAC inherit this, and adapt all consumers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,11 +44,15 @@ contract MyContract is BitmaskRBACPermissionable{
...
+## Usage (Javascript)
+
+Once you have imported your user roles into javascript (e.g. using drizzle), it is super easy to define appropriate constants (e.g. `ADMIN = 1; PUBLISH = 2; VALIDATE = 4`) and then check whether a user bitmask allows a given role or roles. E.g. `(user.role & PUBLISH) !== 0` to check if a user can publish, or even `(user.role & (VALIDATE | ADMIN)) !== 0` to check if a user can Validate or Admin.
+
 ## TODO
 
-Scope what roles a user can add/remove to their role(s)
-Investigate gas use when there are many (> 100) roles, and consider replacing openzeppelin rbac dependency with logic that uses bitmasks natively
-Prevent duplicate roles
+- Ensure we cannot accidentally delete the last user with an `rbac_admin` role, or else the RBAC becomes fixed forever.
```
Comment (gtaschuk) [inline README.md]: this is complete, right?

File: test/utils.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const should = require('chai').should()
```
Comment (gtaschuk) [inline test/utils.js]: I was just looking for something like this today

File: README.md (hunk 1 above)
Comment (nmclrn) [reply README.md]: Yeah. See last two commits for comments on gas usage but if we're both happy with the user counts and admin preservation stuff then this should come out.

File: .circleci/config.yml [hunk 3]
```diff
@@ -19,13 +19,14 @@ jobs:
     working_directory: ~/repo
     docker:
       - image: circleci/node:10.0
-      - image: gtaschuk/testrpc
+      - image: trufflesuite/ganache-cli:latest
+        command: --gasLimit 18000000
```
Comment (nmclrn) [inline .circleci/config.yml]: @gtaschuk This isn't changing the ganache-cli gas limit, which is why the tests are failing. Can you see what I've done wrong?

**Pull request merged**